Sources/NIOHTTP1/HTTPDecoder.swift

```diff
@@ -25,6 +25,7 @@ private struct HTTPParserState {
     var readerIndexAdjustment = 0
     // This is set before http_parser_execute(...) is called and set to nil again after it finish
     var baseAddress: UnsafePointer<UInt8>?
+    var currentError: Error?
```

Comment:
you can save an existential here if you make the type `HTTPParserError`. That probably also save an allocation, maybe ;)

Reply:
actually, only saves an allocation on assignment. But definitely saves a couple of words in this structure if we don't make that an existential type

Reply:
ACK, addressed.